warehouse/query-core/src/main/java/datawave/core/iterators/DatawaveFieldIndexCachingIteratorJexl.java

```diff
@@ -59,6 +59,121 @@
  * 
  */
 public abstract class DatawaveFieldIndexCachingIteratorJexl extends WrappingIterator implements CompositePredicateFilterer {
+    
+    public static abstract class Builder<B extends Builder<B>> {
+        private Text fieldName;
+        protected Text fieldValue;
+        private Predicate<Key> datatypeFilter;
+        private TimeFilter timeFilter;
+        private boolean negated;
+        private PartialKey returnKeyType = DEFAULT_RETURN_KEY_TYPE;
+        private int maxRangeSplit = 11;
+        private FileSystem fs;
+        private Path uniqueDir;
+        private QueryLock queryLock;
+        private boolean allowDirReuse;
+        private long scanThreshold = 10000;
+        private int hdfsBackedSetBufferSize = 10000;
+        private int maxOpenFiles = 100;
+        private boolean sortedUIDs = true;
+        protected QuerySpanCollector querySpanCollector = null;
+        protected volatile boolean collectTimingDetails = false;
+        private volatile long scanTimeout = 1000L * 60 * 60;
+        private Map<String,Map<String,CompositePredicateFilter>> compositePredicateFilters;
+        
+        @SuppressWarnings("unchecked")
+        B self() {
```

Comment:
This is package protected. Is there a reason we'd want to prevent extension into a new package?

Reply:
good catch. i think it should be marked 'protected' so only the Builder subclasses see it